### PR TITLE
fix(obj): fix memory leak in error handling

### PR DESCRIPTION
**Pull request by XuNeo**
### Description of the feature or fix

A clear and concise description of what the bug or new feature is.

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` ([astyle](http://astyle.sourceforge.net/install.html) version [v3.4.12](https://github.com/szepeviktor/astyle/releases/tag/v3.4.12) needs to be installed) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

**Review by C47D — APPROVED** [review]

**Comment by C47D** [comment]
Hi,
Nice catch, it was found by manual review or did you used a tool?

**Comment by XuNeo** [comment]
> Nice catch, it was found by manual review or did you used a tool?

Hi,

It's reported by tool [Coverity](https://scan.coverity.com/).

**Comment by C47D** [comment]
> > Nice catch, it was found by manual review or did you used a tool?
> 
> Hi,
> 
> It's reported by tool [Coverity](https://scan.coverity.com/).

Great, thanks for the information, I didn't knew that tool.

File: src/core/lv_obj_class.c [hunk 1]
```diff
@@ -68,7 +68,10 @@ lv_obj_t * lv_obj_class_create_obj(const lv_obj_class_t * class_p, lv_obj_t * pa
 
         lv_obj_t ** screens = lv_realloc(disp->screens, sizeof(lv_obj_t *) * (disp->screen_cnt + 1));
         LV_ASSERT_NULL(screens);
```
Comment (FASTSHIFT) [inline src/core/lv_obj_class.c]: LV_ASSERT_MALLOC

**Review by FASTSHIFT — APPROVED** [review]

**Review by kisvegabor — APPROVED** [review]
Awesome!

**Pull request merged**